src/docs/development/add-to-app/debugging.md

```diff
@@ -27,19 +27,38 @@ These functionalities are provided by the `flutter attach` mechanism.
 such as through the SDK's CLI tools,
 through VS Code or IntelliJ/Android Studio.
 
+`flutter attach` can connect as soon as you run your `FlutterEngine`, and
+remains attached until your `FlutterEngine` is disposed. But you can invoke
+`flutter attach` before starting your engine. `flutter attach` waits for
+the next available Dart VM that is hosted by your engine.
+
+#### Terminal
+
+Run `flutter attach` or `flutter attach -d deviceId` to attach from the terminal.
+
 {% include app-figure.md image="development/add-to-app/debugging/cli-attach.png" caption="flutter attach via terminal" %}
 
+#### VS Code
+
+Select the correct device using the status bar in VS Code, then run the **Flutter: Attach to Flutter on Device** command from the command palette.
+
 {% include app-figure.md image="development/add-to-app/debugging/vscode-attach.png" caption="flutter attach via VS Code" %}
 
-{% include app-figure.md image="development/add-to-app/debugging/intellij-attach.png" caption="flutter attach via IntelliJ" %}
+Alternatively, create a `.vscode/launch.json` file to enable attaching using the **Run > Start Debugging** command or `F5`:
```

Comment:
Alternatively, create a `.vscode/launch.json` file in your Flutter module project

(since there are 2 projects involved in add-to-app)

Reply:
Yeah, fair point! VS Code users will probably only open the module, but I've made it more explicit.